CryptomatorCommon/Sources/CryptomatorCommonCore/CryptomatorUserDefaults.swift

```diff
@@ -0,0 +1,42 @@
+//
+//  CryptomatorSettings.swift
+//  CryptomatorCommonCore
+//
+//  Created by Tobias Hagemann on 22.09.21.
+//  Copyright © 2021 Skymatic GmbH. All rights reserved.
+//
+
+import CocoaLumberjackSwift
+import Foundation
+
+public protocol CryptomatorSettings {
+	var debugModeEnabled: Bool { get set }
+}
+
+public class CryptomatorUserDefaults {
+	public static let shared = CryptomatorUserDefaults()
+
+	private var defaults = UserDefaults(suiteName: CryptomatorConstants.appGroupName)!
+
+	private func read<T>(property: String = #function) -> T? {
+		defaults.object(forKey: property) as? T
+	}
+
+	private func write<T>(value: T, to property: String = #function) {
+		defaults.set(value, forKey: property)
+		DDLogDebug("Setting \(property) was written with value \(value)")
+	}
+}
+
+extension CryptomatorUserDefaults: CryptomatorSettings {
+	public var debugModeEnabled: Bool {
+		get {
+			#if DEBUG
+			true
+			#else
+			read() ?? false
+			#endif
+		}
+		set { write(value: newValue) }
+	}
```

Comment:
A different approach could be to have a different default value for the debug environment. Something like:

```swift
#if DEBUG
private static let debugModeEnabledDefaultValue = true
#else
private static let debugModeEnabledDefaultValue = false
#endif

public var debugModeEnabled: Bool {
	get { read() ?? CryptomatorUserDefaults.debugModeEnabledDefaultValue }
	set { write(value: newValue) }
}
```

Reply:
Yes, this is indeed a better approach, as it also allows to manually turn off the debug mode for debug builds.
I have updated that part of the code, thank you.